Schema/app/006-merge-account-function.blue.sql

```diff
@@ -0,0 +1,42 @@
+INSERT INTO odin.migration VALUES('app', '006-merge-account-function.blue.sql');
+
+ALTER TABLE odin.app_user_role_ledger
+    DROP CONSTRAINT app_user_role_ledger_app_id_fkey,
+    ADD CONSTRAINT app_user_role_ledger_app_id_fkey
+        FOREIGN KEY (app_id)
+        REFERENCES odin.app (app_id) MATCH SIMPLE
+        ON UPDATE NO ACTION ON DELETE NO ACTION DEFERRABLE,
+    ADD CONSTRAINT app_user_role_ledger_identity_id_fkey
+        FOREIGN KEY (identity_id)
+        REFERENCES odin.identity_record (id) MATCH SIMPLE
+        ON UPDATE NO ACTION ON DELETE NO ACTION DEFERRABLE;
+
+CREATE FUNCTION odin.merge_account_app(merge_from TEXT, merge_to TEXT)
+RETURNS VOID AS
+$body$
+DECLARE
+  merge_to_installation_id TEXT;
+  reference_val TEXT = 'merge:'||merge_from||':'||merge_to;
+  annotation JSON = json_build_object('merge_from', merge_from,'merge_to', merge_to);
+BEGIN
+  SELECT installation_id INTO merge_to_installation_id FROM odin.app_user WHERE identity_id = merge_to LIMIT 1;
+
+  INSERT INTO odin.app_user_ledger (reference, app_id, identity_id, annotation)
```

Comment:
We will not insert into ledgers table but update the actual data tables instead.

Reply:
This should try to update first, if cannot, we do the delete